src/mscorlib/corefx/System/Runtime/InteropServices/StringBuffer.cs

```diff
@@ -252,29 +236,24 @@ public void Append(StringBuffer value, uint startIndex, uint count)
         /// of <paramref name="value"/> characters.
         /// </exception>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="destination"/> is null.</exception>
-        public unsafe void CopyTo(uint bufferIndex, StringBuffer destination, uint destinationIndex, uint count)
+        public void CopyTo(int bufferIndex, ref StringBuffer destination, int destinationIndex, int count)
         {
-            if (destination == null) throw new ArgumentNullException(nameof(destination));
             if (destinationIndex > destination._length) throw new ArgumentOutOfRangeException(nameof(destinationIndex));
             if (bufferIndex >= _length) throw new ArgumentOutOfRangeException(nameof(bufferIndex));
             if (_length < checked(bufferIndex + count)) throw new ArgumentOutOfRangeException(nameof(count));
 
             if (count == 0) return;
-            uint lastIndex = checked(destinationIndex + count);
-            if (destination._length < lastIndex) destination.Length = lastIndex;
-
-            Buffer.MemoryCopy(
-                source: CharPointer + bufferIndex,
-                destination: destination.CharPointer + destinationIndex,
-                destinationSizeInBytes: checked((long)(destination.ByteCapacity - (destinationIndex * sizeof(char)))),
-                sourceBytesToCopy: checked((long)count * sizeof(char)));
+            int lastIndex = checked(destinationIndex + count);
+            if (destination.Length < lastIndex) destination.Length = lastIndex;
+
+            Array.Copy(UnderlyingArray, bufferIndex, destination.UnderlyingArray, destinationIndex, count);
```

Comment:
The difference between the two is neglible. I think Array.Copy is better here given the context - more readable code.